frontend/packages/console-shared/src/hooks/useBoundingClientRect.ts

```diff
@@ -0,0 +1,23 @@
+import * as React from 'react';
+import { useResizeObserver } from './useResizeObserver';
+
+type BoundingClientRect = ClientRect | null;
+
+export const useBoundingClientRect = (targetElement: HTMLElement | null): BoundingClientRect => {
+  const [clientRect, setClientRect] = React.useState<BoundingClientRect>(null);
+
+  const observerCallback = React.useCallback(() => {
+    setClientRect(targetElement ? targetElement.getBoundingClientRect() : null);
+  }, [targetElement]);
+
+  React.useEffect(() => {
+    if (targetElement) {
+      setClientRect(targetElement.getBoundingClientRect());
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
+
+  useResizeObserver(observerCallback);
```

Comment:
I think we need a new issue to move to using popperjs for the static spotlight as well. There's also another issue with the way the see through spotlight is rendered in that it affects the colors due to the css effect.

For now the resize observer is ok.